src/Cech_complex/include/gudhi/MEB_filtration.h

```diff
@@ -0,0 +1,112 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Marc Glisse
+ *
+ *    Copyright (C) 2023 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef MEB_FILTRATION_H_
+#define MEB_FILTRATION_H_
+
+namespace Gudhi::cech_complex {
+
+/**
+ * \ingroup cech_complex
+ *
+ * \brief
+ * Given a simplicial complex and an embedding of its vertices, this assigns to
+ * each simplex a filtration value equal to the squared radius of its minimal
+ * enclosing ball.
+ *
+ * Applied on a Cech complex, it recomputes the same values (squared). Applied on a Delaunay triangulation, it computes the Delaunay-Cech filtration.
+ *
+ * \tparam Kernel CGAL kernel: either Epick_d or Epeck_d.
+ * \tparam PointRange Random access range of `Kernel::Point_d`.
+ *
+ * @param[in] k The geometric kernel.
+ * @param[in] complex The simplicial complex.
+ * @param[in] points Embedding of the vertices of the complex.
+ * @param[in] exact If true and `Kernel` is <a href="https://doc.cgal.org/latest/Kernel_d/structCGAL_1_1Epeck__d.html">CGAL::Epeck_d</a>, the filtration values are computed exactly. Default is false.
+ */
+
+template<typename Kernel, typename SimplicialComplexForMEB, typename PointRange>
+void assign_MEB_filtration(Kernel&&k, SimplicialComplexForMEB& complex, PointRange points, bool exact = false) {
+  using Point_d = typename Kernel::Point_d;
+  using FT = typename Kernel::FT;
+  using Sphere = std::pair<Point_d, FT>;
+
+  using Vertex_handle = typename SimplicialComplexForMEB::Vertex_handle;
+  using Simplex_handle = typename SimplicialComplexForMEB::Simplex_handle;
+  using Filtration_value = typename SimplicialComplexForMEB::Filtration_value;
+
+  std::vector<Sphere> cache_;
+  std::vector<Point_d> pts;
+  CGAL::NT_converter<FT, Filtration_value> cvt;
+
+  auto fun = [&](Simplex_handle sh, int dim){
+    if (dim == 0) complex.assign_filtration(sh, 0);
+    else if (dim == 1) {
+      // For a Simplex_tree, this would be a bit faster, but that's probably negligible
+      // Vertex_handle u = sh->first; Vertex_handle v = self_siblings(sh)->parent();
+      auto verts = complex.simplex_vertex_range(sh);
+      auto vert_it = verts.begin();
+      Vertex_handle u = *vert_it;
+      Vertex_handle v = *++vert_it;
+      auto&& pu = points[u];
+      Point_d m = k.midpoint_d_object()(pu, points[v]);
+      FT r = k.squared_distance_d_object()(m, pu);
+      if (exact) CGAL::exact(r);
+      complex.assign_key(sh, cache_.size());
+      complex.assign_filtration(sh, std::max(cvt(r), Filtration_value(0)));
```

Comment:
IIUC, this means that `squared_distance_d_object` can return negative values ?

Reply:
With exact computations, no, but then we wouldn't need a lot of other code (see [1488c9b](https://github.com/GUDHI/gudhi-devel/pull/919/commits/1488c9bed79a0371cff48f12f51db0ea8a6ac840)).
Without rounding errors... it still seem very unlikely to get a negative result, but better safe than sorry, especially since I expect this to take a negligible amount of time.